src/segment-loader.js

```diff
@@ -423,15 +423,71 @@ export const shouldFixBadTimelineChanges = (timelineChangeController) => {
   return false;
 };
 
+/**
+ * Fixes certain bad timeline scenarios by resetting the loader.
+ *
+ * @param {SegmentLoader} segmentLoader
+ */
 export const fixBadTimelineChange = (segmentLoader) => {
   if (!segmentLoader) {
     return;
   }
+
   segmentLoader.pause();
   segmentLoader.resetEverything();
   segmentLoader.load();
 };
 
+/**
+ * Check if the pending audio timeline change is behind the
+ * pending main timeline change.
+ *
+ * @param {SegmentLoader} segmentLoader
+ * @return {boolean}
+ */
+const isAudioTimelineBehind = (segmentLoader) => {
+  const pendingAudioTimelineChange = segmentLoader.timelineChangeController_.pendingTimelineChange({ type: 'audio' });
+  const pendingMainTimelineChange = segmentLoader.timelineChangeController_.pendingTimelineChange({ type: 'main' });
+  const hasPendingTimelineChanges = pendingAudioTimelineChange && pendingMainTimelineChange;
+
+  if (hasPendingTimelineChanges && pendingAudioTimelineChange.to < pendingMainTimelineChange.to) {
+    return true;
+  }
+
+  return false;
```

Comment:
you can simply:
  `return hasPendingTimelineChanges && pendingAudioTimelineChange.to < pendingMainTimelineChange.to;`